packages/slider/src/mwc-slider.ts

```diff
@@ -0,0 +1,208 @@
+    const {value, min, max, step, disabled, discrete, markers, _numMarkers} = this;
+    const hostClassInfo = {
+      'mdc-slider--discrete': discrete,
+      'mdc-slider--display-markers': markers && discrete,
+    };
+    return html`
+      ${this.renderStyle()}
+      <div class="mdc-slider ${classMap(hostClassInfo)}" tabindex="0" role="slider"
+        aria-valuemin="${min}" aria-valuemax="${max}" aria-valuenow="${value}"
+        aria-disabled="${disabled}" data-step="${step}">
+      <div class="mdc-slider__track-container">
+        <div class="mdc-slider__track"></div>
+        ${discrete && markers ? html`<div class="mdc-slider__track-marker-container">
+          ${repeat(new Array(_numMarkers), () => html`<div class="mdc-slider__track-marker"></div>`)}
+        </div>` : ''}
+      </div>
+      <div class="mdc-slider__thumb-container">
+        ${discrete ? html`<div class="mdc-slider__pin">
+          <span class="mdc-slider__pin-value-marker"></span>
+        </div>` : ''}
+        <svg class="mdc-slider__thumb" width="21" height="21">
+          <circle cx="10.5" cy="10.5" r="7.875"></circle>
+        </svg>
+        <div class="mdc-slider__focus-ring"></div>
+      </div>
+    </div>`;
+  }
+
+  protected createAdapter() {
+    return {
+      ...super.createAdapter(),
+      getAttribute: (name: string) => this.mdcRoot.getAttribute(name),
+      setAttribute: (name: string, value: string) => this.mdcRoot.setAttribute(name, value),
+      removeAttribute: (name: string) => this.mdcRoot.removeAttribute(name),
+      computeBoundingRect: () => this.mdcRoot.getBoundingClientRect(),
+      getTabIndex: () => this.mdcRoot.tabIndex,
+      registerInteractionHandler: (type: string, handler: EventListener) =>
+        this.mdcRoot.addEventListener(type, handler),
+      deregisterInteractionHandler: (type: string, handler: EventListener) =>
+        this.mdcRoot.removeEventListener(type, handler),
+      registerThumbContainerInteractionHandler: (type: string, handler: EventListener) =>
+        this.thumbContainer.addEventListener(type, handler),
+      deregisterThumbContainerInteractionHandler: (type: string, handler: EventListener) =>
+        this.thumbContainer.removeEventListener(type, handler),
+      registerBodyInteractionHandler: (type: string, handler: EventListener) =>
+        document.body.addEventListener(type, handler),
+      deregisterBodyInteractionHandler: (type: string, handler: EventListener) =>
+        document.body.removeEventListener(type, handler),
+      registerResizeHandler: (handler: EventListener) =>
+        window.addEventListener('resize', handler),
+      deregisterResizeHandler: (handler: EventListener) =>
+        window.removeEventListener('resize', handler),
+      notifyInput: () => {
+        this.value = this.mdcFoundation.getValue();
+        this.dispatchEvent(new CustomEvent(INPUT_EVENT, {detail: this, bubbles: true, cancelable: true}));
+      },
+      notifyChange: () => {
```

Comment:
Verify that this does not double-notify since setting `this.value` result in setting `setValue` on the foundation (which may cause this notify to go again).